### PR TITLE
Add test_lite custom command for parallel test running.

### PR DESCRIPTION
**Pull request by jtattermusch**
Allows running multiple tests in parallel (by preventing interference between egg fetches or build steps).

Fixes https://github.com/grpc/grpc/issues/5568.

### CODE REVIEW

**Comment by jtattermusch** [comment]
cc @makdharma

**Comment by jtattermusch** [comment]
Let's review & merge soon so we can get rid of flakes causes by run_tests.py and focus on the real thing.

File: src/python/grpcio/commands.py [hunk 1]
```diff
@@ -264,6 +264,41 @@ def run(self):
...
+  def run(self):
+    self._add_eggs_to_path()
+
+    import tests
+    loader = tests.Loader()
+    loader.loadTestsFromNames(['tests'])
+    runner = tests.Runner()
+    result = runner.run(loader.suite)
+    if not result.wasSuccessful():
+      sys.exit(1)
```
Comment (soltanmm) [inline src/python/grpcio/commands.py]: Could we change this to `sys.exit('Test failure')`? If we need to make it any more detailed we can do so, but right now I'd rather like to see Python provide, independent of the runner behavior, a little bit more than just an error code on exit.

File: src/python/grpcio/commands.py [hunk 2]
```diff
@@ -264,6 +264,41 @@ def run(self):
...
+
+  def _add_eggs_to_path(self):
```
Comment (soltanmm) [inline src/python/grpcio/commands.py]: And I guess we'll just have to clean this up later. ¯\_(ツ)_/¯

**Comment by soltanmm** [comment]
Python failures are known flakes.

**Pull request merged**